### ISSUE TITLE
`kubectl get events --watch` appears borked with non-default output formats

### ISSUE THREAD

**Issue by DazWilkin** [kubernetes/kubectl#542]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):

`is:issue "kubectl get events"`

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kubernetes version** (use `kubectl version`): v1.10.7-gke.2

**Environment**:
- **Cloud provider or hardware configuration**: Kubernetes Engine
- **OS** (e.g. from /etc/os-release): COS
- **Kernel** (e.g. `uname -a`): 4.14.56+
- **Install tools**:
- **Others**:

**What happened**:

```
kubectl get events --all-namespaces --watch --output=jsonpath="{.message}"
kubectl get events --all-namespaces --watch --output=json
kubectl get events --all-namespaces --watch --output=yaml
kubectl get events --all-namespaces --watch --output=custom-columns=MESSAGE:.message
```
(Appears to) correctly stream a series of recent events, then blocks awaiting subsequent events. On the occurrence of the next event, the command terminates.

**What you expected to happen**:

I expect the command to behave consistently with+without the output specifier (`output=wide` works). 

```
kubectl get events --all-namespaces --watch
kubectl get events --all-namespaces --watch --output=wide
```
I had originally been using a `range` command in the JSON output but realized this probably is erroneous (non-sensical) when the events are being watched:
```
kubectl get events --output=jsonpath="{range .items[*]}{.message}\n{end}" --watch
```

**How to reproduce it** (as minimally and precisely as possible):
See examples above.

**Anything else we need to know**:
A reason for wanting to do this is -- as is shown -- to be able more conveniently filter the output results. In this case, I'm playing around with the node-problem-detector and I'm only interested in seeing the `message` value in order to determine that the utility is working.

**Comment by m1kola** [comment]
I just built `kubectl` from master (from the commit [`6a8a859`](https://github.com/kubernetes/kubernetes/commit/6a8a8597f5bf4de29b7b88384529188df0b71337)) and it seems to work well. Also v1.12.0 works as expected.

I can, however, reproduce it with v1.11.2. So, I guess, the issue was fixed somewhere in between v1.11.2 and v1.12.0. I didn't manage to find the relevant pull request, though.

Tested on darwin with a GKE cluster v1.10.6-gke.2

@DazWilkin could you, please, confirm that Kubectl 1.12.0 works for you as expected? If you are still experiencing it with the newever version of kubectl, please, include as much details about your set up as possible (for example, output of `kubectl version` will be very helpful).

**Comment by DazWilkin** [comment]
You bet! I'm taking a week's leave but will try this when I'm back on the 8th. Thanks!

**Comment by czka** [comment]
@m1kola 
you wrote:
>  the issue was **fixed** somewhere in between v1.11.2 and v1.12.0

You meant **introduced**, right? Asking this cause I've been debugging the issue myself and have a problem identifying an unaffected version. Eg. `get events --watch --output=json` in `v1.13.0` seems to work OK, most of the time, but it would terminate at random now and then.

**Comment by pswica** [comment]
@m1kola I could not reproduce as well. I think this is good to close.

```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T16:23:09Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T16:14:56Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```
